### ISSUE TITLE
I can’t install Atom - newbie

### ISSUE THREAD

**Issue by Jessica-Ferreira** [atom/atom#4583]
Hi, guys.
I've started to learn how to code and found out about Atom. I downloaded the installer and when I run it shows a error message "Instalation has failed/Failed to extract installer", even if I run as admin. Downloaded the installer twice. Am I doing something wrong? My os is win8 x64.
Thanks.

**Comment by indcoder** [comment]
![screenshot 2015-01-14 15 04 54](https://cloud.githubusercontent.com/assets/2099300/5736863/db2e3af2-9c00-11e4-85b6-dc1b72869503.png)

The issue is existing in the latest installer still. Any ETA for this?

**Comment by 50Wliu** [comment]
Could you please click on `Open Setup Log` and paste its contents here?

**Comment by indcoder** [comment]
@50Wliu When I click, it opens a directory but there is no log within that directory

**Comment by michaeljcalkins** [comment]
Windows 7 x64

I just downloaded the latest binary and this error came up.

```
Program: Starting Squirrel Updater: --install .
Program: Starting install, writing to C:\Users\Michael J. Calkins\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Reading RELEASES file from C:\Users\Michael J. Calkins\AppData\Local\SquirrelTemp
ApplyReleasesImpl: Writing files to app directory: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0
IEnableLogger: Failed to write file: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
   at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__80(IPackageFile x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__7f()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__86.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()
```

**Comment by michaeljcalkins** [comment]
When I try again running as administrator:

```
Program: Starting Squirrel Updater: --install .
Program: Starting install, writing to C:\Users\Michael J. Calkins\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Reading RELEASES file from C:\Users\Michael J. Calkins\AppData\Local\SquirrelTemp
ApplyReleasesImpl: Found partially applied release folder, killing it: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\locales
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\.bin
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\bin
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\native-module
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\native-module\src
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\.bin
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\async
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\async\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\colors
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\exit
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\colors\themes
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\exit\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Loca
atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\harmony-collections
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\mixto
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\mixto\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\lib
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\node_modules
Utility: Starting to delete folder: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\node_modules\harmony-collections
ApplyReleasesImpl: Writing files to app directory: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0
IEnableLogger: Failed to write file: C:\Users\Michael J. Calkins\AppData\Local\atom\app-0.174.0: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
   at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__80(IPackageFile x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__7f()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__86.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()
```

**Comment by kevinsawicki** [comment]
@michaeljcalkins Would you mind opening a new issue for that here? https://github.com/Squirrel/Squirrel.Windows/issues/new

**Comment by michaeljcalkins** [comment]
@kevinsawicki Opened here https://github.com/Squirrel/Squirrel.Windows/issues/192

**Comment by paulcbetts** [comment]
@kevinsawicki This is an Atom bug - his username is too long for us to be able to install Atom in LocalAppData

**Comment by patrickse** [comment]
Facing the same problem...

**Comment by CameronAM** [comment]
I'm hitting the same issue.

```
Program: Starting Squirrel Updater: --install . --silent
Program: Starting install, writing to C:\Users\cameron.morrison\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.FileNotFoundException: Could not find file 'C:\Users\cameron.morrison\AppData\Local\atom\packages\RELEASES'.
File name: 'C:\Users\cameron.morrison\AppData\Local\atom\packages\RELEASES'
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__28.MoveNext()
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\packages
CheckForUpdateImpl: Reading RELEASES file from C:\Users\cameron.morrison\AppData\Local\SquirrelTemp
CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
ApplyReleasesImpl: Found partially applied release folder, killing it: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\locales
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\.bin
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\bin
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\native-module
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\native-module\src
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\.bin
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\async
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\async\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\colors
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\colors\themes
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\exit
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\exit\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\harmony-collections
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\mixto
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\mixto\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\lib
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\node_modules
Utility: Starting to delete folder: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0\resources\app\apm\node_modules\atom-package-manager\node_modules\first-mate\node_modules\emissary\node_modules\property-accessors\node_modules\harmony-collections
ApplyReleasesImpl: Writing files to app directory: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0
IEnableLogger: Failed to write file: C:\Users\cameron.morrison\AppData\Local\atom\app-0.174.0: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
   at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__80(IPackageFile x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__7f()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__86.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()

```

**Comment by 50Wliu** [comment]
@CameronAM please comment on #5109-that's the issue specifically for your bug.

**Comment by CSharpFan** [comment]
Could the problem be the less-compile-cache? That one has some long path names.

**Comment by CSharpFan** [comment]
Update: I ran installer, got an error (no screenshot, sorry):

```text
Program: Starting Squirrel Updater: --install .
Program: Starting install, writing to C:\Users\Foo Bar\AppData\Local\SquirrelTemp
CheckForUpdateImpl: Failed to load local releases, starting from scratch: System.IO.DirectoryNotFoundException: Could not find a part of the path 'C:\Users\Foo Bar\AppData\Local\atom\packages\RELEASES'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Squirrel.Utility.LoadLocalReleases(String localReleaseFile)
   at Squirrel.UpdateManager.CheckForUpdateImpl.<CheckForUpdate>d__28.MoveNext()
CheckForUpdateImpl: Reading RELEASES file from C:\Users\Foo Bar\AppData\Local\SquirrelTemp
CheckForUpdateImpl: First run or local directory is corrupt, starting from scratch
ApplyReleasesImpl: Writing files to app directory: C:\Users\Foo Bar\AppData\Local\atom\app-0.177.0
IEnableLogger: Failed to write file: C:\Users\Foo Bar\AppData\Local\atom\app-0.177.0: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
IEnableLogger: Failed to install package to app dir: System.IO.PathTooLongException: The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
   at System.IO.PathHelper.GetFullPathName()
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength, Boolean expandShortPaths)
   at System.IO.Path.NormalizePath(String path, Boolean fullCheck, Int32 maxPathLength)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at NuGet.PhysicalPackageFile.GetStream()
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass8b.<copyFileToLocation>b__8a()
   at Squirrel.Utility.LogIfThrows(IFullLogger This, LogLevel level, String message, Action block)
   at Squirrel.UpdateManager.ApplyReleasesImpl.copyFileToLocation(FileSystemInfo target, IPackageFile x)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__80(IPackageFile x)
   at System.Collections.Generic.List`1.ForEach(Action`1 action)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<>c__DisplayClass84.<installPackageToAppDir>b__7f()
   at System.Threading.Tasks.Task.InnerInvoke()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.UpdateManager.ApplyReleasesImpl.<installPackageToAppDir>d__86.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Squirrel.Utility.<LogIfThrows>d__4b`1.MoveNext()
````

Notice the error is different the second time where it first tries to clear the atom\app-.01770 folder.

I then extracted the Atom installer and manually copied the contents to `C:\Users\Foo Bar\AppData\Local\atom\app-0.177.0`

App starts :)

**Comment by 50Wliu** [comment]
@CSharpFan see #5109 for details.

I'm going to close this bug because it seems a bit generic and issues aren't going to where they should be going.

**Comment by paulcbetts** [comment]
@CSharpFan Hm, now that's interesting.

**Comment by jcs224** [comment]
Having this same issue now as well. I do happen to run this on a corporate computer.

**Comment by paulcbetts** [comment]
@jcs224 Do you have the *same* issue, or do you just have *an* issue? Can you paste in your log file?

**Comment by jcs224** [comment]
Hey @paulcbetts , sorry to cause all this commotion, I also posted on thread #5881. The issue on *this* thread actually matches my issue, "Failed to extract installer" rather than a generic error message.

Unfortunately, same deal, no log file present. Just opens up my Documents folder. Permissions issue, perhaps? Maybe I should check that.

I will note that a co-worker just installed Atom on his machine a few minutes ago, near-identical setup and it works fine, so my issue might be more unique to me.

**Comment by pingween** [comment]
I ran into this issue (trying to install with cocolately first, then running the downloaded setup directly) with the same results - no log file, nothing useful in `%LocalAppData%\SquirrelTemp`.

I tried to run update.exe from the command line and it turns out the Active directory Group Policy blockes `update.exe` (company notebook), but it only tells when you run directly `update.exe`. You might wanna check this with your system administrator if that's the case.

A possible workaround is to rename the nupkg to a .zip and unzip `lib\net45`'s contents to `%LocalAppData%\atom\app-{version}`.

Due to the Group Policies it won't update automatically anyhow, so, it works for me.

**Comment by mikememo** [comment]
For me, Group Policy blocks any executables running from AppData, temp folders, Downloads, etc.

**Comment by paulcbetts** [comment]
@mikememo Oh yeah, don't do that :)

**Comment by mikememo** [comment]
@paulcbetts Corporate GP -- it's saved our butts several times :)  (notably, cryptolocker!)

**Comment by stajs** [comment]
I've hit the Group Policy block too (not with Atom, but another Squirrel-based installer):

```
C:\Users\<Name>\AppData\Local\SquirrelTemp>Update.exe --install. --packagesDir .\<Package>.nupkg
This program is blocked by group policy. For more information, contact your system administrator.
```

**Comment by 50Wliu** [comment]
@stajs @paulcbetts has stated that he might work on an installer that will write to Program Files instead, but it'll be in the distant future.
***
I'm locking this issue because it's drawn a lot of generic "the installer isn't working" comments when usually it's better to create a new issue or find a recently-closed/still-open issue to comment on.

If you have an issue with the installer, first please check for duplicates using the [installer](https://github.com/atom/atom/issues?q=is%3Aissue+label%3Ainstaller+sort%3Aupdated-desc) label.  If there's nothing that matches your error, feel free to create a new issue containing the following info:
* Your Atom version
* The Squirrel setup log
* Your Windows version
* Whether or not you're behind a proxy
* Whether or not your company is restricting installations through group policy (this should be reported automatically in future versions of Atom)
* What antiviruses you're running
* Any other info that you feel like might be useful

Thanks! :smile:

**Comment by paulcbetts** [comment]
> Whether or not your company is restricting installations through group policy (this should be reported automatically in future versions of Atom)

This is tough if we can't run any code on the box :(

**Comment by 50Wliu** [comment]
Sorry, misread one of your Squirrel commits.  Updated slightly :wink:.
